lib/locations/locations.go

```diff
@@ -123,23 +131,42 @@ func defaultConfigDir() string {
 		return filepath.Join(os.Getenv("AppData"), "Syncthing")
 
 	case "darwin":
-		dir, err := fs.ExpandTilde("~/Library/Application Support/Syncthing")
-		if err != nil {
-			fmt.Println(err)
-			panic("Failed to get default config dir")
-		}
-		return dir
+		return filepath.Join(home, "Library/Application Support/Syncthing")
 
 	default:
 		if xdgCfg := os.Getenv("XDG_CONFIG_HOME"); xdgCfg != "" {
 			return filepath.Join(xdgCfg, "syncthing")
 		}
-		dir, err := fs.ExpandTilde("~/.config/syncthing")
-		if err != nil {
-			fmt.Println(err)
-			panic("Failed to get default config dir")
+		return filepath.Join(home, ".config/syncthing")
+	}
+}
+
+// defaultDataDir returns the default data directory, which usually is the
+// config directory but might be something else.
+func defaultDataDir(home, config string) string {
+	switch runtime.GOOS {
+	case "windows", "darwin":
+		return config
+
+	default:
+		// If a database exists at the "normal" location, use that anyway.
+		if _, err := os.Lstat(filepath.Join(home, "syncthing", databaseDirname)); err == nil {
+			return config
```

Comment:
This is weird though as we stat one location and return another. What if I have one instance running with the database in the default location and start another with `-home ~/asdasfd` (I do this all the time)

Reply:
It's not just weird, it's plain wrong. It should `filepath.Join(config, databaseDirname)` in the stat.

And `-home` overrules default locations, i.e. it doesn't matter what happens in this function in that case.

Also let me do 's/home/userHome/' - it's annoying, but it's definitely less error prone than calling a variable `home` that's not connected with `-home`.

Reply:
Yes please